diff --git a/CMakeLists.txt b/CMakeLists.txt
index ab399a69..6ecd2f24 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -407,13 +407,14 @@ set(EXAMPLE_CONFIG_FILES
 # Configuration parameters
 ################################################################################
 if(${CMAKE_SYSTEM_NAME} MATCHES "QNX")
-    set (VSOMEIP_BASE_PATH "/var")
+  set(VSOMEIP_BASE_PATH_DEFAULT "/var")
 else()
-    set (VSOMEIP_BASE_PATH "/tmp")
+  set(VSOMEIP_BASE_PATH_DEFAULT "/tmp")
 endif()
-if (BASE_PATH)
-set (VSOMEIP_BASE_PATH ${BASE_PATH})
-endif ()
+set(VSOMEIP_BASE_PATH
+    ${VSOMEIP_BASE_PATH_DEFAULT}
+    CACHE PATH "Base path for vsomeip sockets"
+)
 
 set (VSOMEIP_DIAGNOSIS_ADDRESS "0x01")
 if (DIAGNOSIS_ADDRESS)
